Add Caddx/GE/Interlogix NetworX alarm panel support

kk7ds opened the pull request:
This adds support for NetworX-based alarm panels and should work for
any such panel equipped with a NX584 serial interface module. This
includes NX-4/6/8/8E, where the NX8E has this interface built-in.

It requires the pynx584 module, and requires running the server
component somewhere that has connectivity to the panel via serial,
which may include a serial-over-lan connection.

kk7ds committed:
Add Caddx/GE/Interlogix NetworX alarm panel support

This adds support for NetworX-based alarm panels and should work for
any such panel equipped with a NX584 serial interface module. This
includes NX-4/6/8/8E, where the NX8E has this interface built-in.

It requires the pynx584 module, and requires running the server
component somewhere that has connectivity to the panel via serial,
which may include a serial-over-lan connection.

balloob commented on homeassistant/components/alarm_control_panel/nx584.py:
```diff
@@ -0,0 +1,104 @@
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Setup nx584. """
+    host = config.get('host', 'localhost:5007')
+
```
Can you verify the login is correct?

kk7ds replied:
There's no login, just connectivity. But I can test connectivity in __init__ and log something, yeah.

balloob commented on homeassistant/components/alarm_control_panel/nx584.py:
```diff
@@ -0,0 +1,104 @@
+                                 STATE_ALARM_ARMED_AWAY)
+import homeassistant.components.alarm_control_panel as alarm
+
+REQUIREMENTS = ['pynx584']
```
Make sure you specify a version.

balloob commented on homeassistant/components/alarm_control_panel/nx584.py:
```diff
@@ -0,0 +1,104 @@
+
+    def alarm_disarm(self, code=None):
+        self._alarm.disarm(code)
+        self.update_ha_state()
```
Instead of adding these `update_ha_state` calls after this and the other methods, can you fix it in the camera component. This should have been there already.

After [this line](https://github.com/balloob/home-assistant/blob/dev/homeassistant/components/alarm_control_panel/__init__.py#L63) with same indent as `getattr`:
```python
if camera.should_poll:
    camera.update_ha_state(True)
```

kk7ds replied:
Hmm, I'm not sure I know what you're saying. Did you mean::

```
 if self.should_poll:
     self.update_ha_state()
```

?

kk7ds replied:
Nevermind, I got it :)

kk7ds committed:
Fix alarm service handler state updates

This changes the service handler dispatch code to always
call update_ha_state(), and cleans up the alarm platforms that
were calling it themselves.

balloob commented on homeassistant/components/alarm_control_panel/nx584.py:
```diff
@@ -0,0 +1,106 @@
+            self._alarm = client.Client('http://%s' % host)
+            self._alarm.list_zones()
+        except requests.exceptions.ConnectionError as ex:
+            _LOGGER.error('Unable to connect to NX584: %s',
```
The alarm control panel component will keep polling this device because it has been added on line 38. You could consider catching the exception in `setup_platform` so that it will not get added to HA.

kk7ds replied:
Sure.
